dlp/snippets/inspect/inspect_file.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2018 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package inspect
+
+// [START dlp_inspect_file]
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+
+	"cloud.google.com/go/dlp/apiv2"
+	dlppb "google.golang.org/genproto/googleapis/privacy/dlp/v2"
+)
+
+func inspectFile(projectID, filepath, fileType string) error {
+	// projectID := "my-project-id";
+	// filePath := "path/to/image.png";
+	// fileType := "IMAGE"
+
+	// Initialize client
+	client, err := dlp.NewClient(context.Background())
+	if err != nil {
+		return err
+	}
+	defer client.Close() // Closing the client safely stops any background threads or connections.
+
+	// Prepare the request
+	// Set the item for the request.
+	data, err := ioutil.ReadFile(filepath)
+	if err != nil {
+		return err
+	}
+	var itemType dlppb.ByteContentItem_BytesType
+	switch fileType {
+	case "IMAGE":
+		itemType = dlppb.ByteContentItem_IMAGE
+	case "TEXT_UTF8":
+		itemType = dlppb.ByteContentItem_TEXT_UTF8
+	default:
+		return errors.New(fmt.Sprintf("invalid ByteType for ByteContentItem: '%s'", fileType))
+	}
+	item := &dlppb.ContentItem{
+		DataItem: &dlppb.ContentItem_ByteItem{
+			ByteItem: &dlppb.ByteContentItem{
+				Type: itemType,
+				Data: data,
+			},
+		},
+	}
+	// Set the required InfoTypes for the inspection config.
+	var infoTypes []*dlppb.InfoType
+	for _, it := range []string{"PHONE_NUMBER", "EMAIL_ADDRESS", "CREDIT_CARD_NUMBER"} {
+		infoTypes = append(infoTypes, &dlppb.InfoType{Name: it})
+	}
+	// Set the inspection configuration for the request.
+	config := &dlppb.InspectConfig{
+		InfoTypes:    infoTypes,
+		IncludeQuote: true,
+	}
+
+	// Create and send the request.
+	req := &dlppb.InspectContentRequest{
+		Parent:        "projects/" + projectID,
+		Item:          item,
+		InspectConfig: config,
```

Comment:
I much prefer inlined proto fields rather than creating many variables.

Only use variables if necessary.

Reply:
Additionally, I've run into issues with samples like this before where there were existing variables with those names. This sample is not copy-pasteable if the code already has an "item" or "config" variable in scope.